java-vtl-script/src/main/java/no/ssb/vtl/script/VTLScriptEngine.java

```diff
@@ -134,89 +115,57 @@ public Object eval(String script, ScriptContext context) throws ScriptException
         return eval(new StringReader(script), context);
     }
 
-    @Override
-    public Object eval(Reader reader, ScriptContext context) throws ScriptException {
-        /*
-            Until compilation is done, this is the main method that allows
-            VTL execution.
-            Exceptions' hierarchy is as follow:
-                - ScriptException
-                    - CompilationException
-                        - SyntaxException
-                        - TypeException
-                        - ConstraintException
-                    - ValidationException
-                    - VTLRuntimeException
-            The WrappedScriptException is used to report errors that are not originating
-            from the VTL Parser.
-         */
-        try {
-            VTLLexer lexer = new VTLLexer(new ANTLRInputStream(reader));
-            VTLParser parser = new VTLParser(new CommonTokenStream(lexer));
-
-            lexer.removeErrorListeners();
-            parser.removeErrorListeners();
-
-            BaseErrorListener errorListener = new BaseErrorListener() {
-                @Override
-                public void syntaxError(Recognizer<?, ?> recognizer, Object offendingSymbol, int line, int column, String msg, RecognitionException e) {
-                    WrappedException wrappedException;
-                    if (e instanceof WrappedException) {
-                        wrappedException = (WrappedException) e;
-                    } else {
-                        wrappedException = new WrappedException(
-                                msg,
-                                recognizer,
-                                e != null ? e.getInputStream() : null,
-                                e != null ? (ParserRuleContext) e.getCtx() : null,
-                                new SyntaxException(
-                                        msg, null, line, column, "VTL-0199"
-                                )
-                        );
-                    }
-                    wrappedException.setLine(line);
-                    wrappedException.setColumn(column);
-                    throw new ParseCancellationException(msg, wrappedException);
-                }
-            };
-            lexer.addErrorListener(errorListener);
-            parser.addErrorListener(errorListener);
+    public VTLParser.StartContext compile(Reader reader, Consumer<VTLScriptException> errorConsumer) throws IOException {
```

Comment:
I'm not sure about the `compile` word here. In antlr it is used in the context of compiling a grammar file.

Reply:
right, how about validate?

Reply:
maybe `getRootContext` or `getStartContext`. Or maybe split this method as it sets up lexer and parser and also gets the start context.